### ISSUE TITLE
sqlite connector

### ISSUE THREAD

**Issue by nicgrobler** [influxdata/flux#1933]
Hi,

I have been playing with Flux and extended your "sql" package to include sqlite3 - so it supports:  
- sql.from()
- sql.to()  

If interested, and you are not already doing it, I am happy to submit a PR.

I added it because SQLite is an extremely useful "medium" which many platforms support without the need for a full-blown relational DB, with server.

**Comment by nathanielc** [comment]
@nicgrobler We would love to see a PR, we are not actively working to add sqlite support but see it as a valuable addition. Thanks. Let us know if you get stuck on anything.

**Comment by nicgrobler** [comment]
@nathanielc thanks - couple of questions have come up, which need some strategic decisions from you guys that go way beyond simply adding SQLite (I have 2 working branches, but they are very different, for the reasons below):
 
1. As most flavours of "sql DB" out there do not support boolean types (essentially stored as flavours of INT)we need to decide whether we support "SQL booleans", or not: Saying "No" means that our own code will be very much less ambiguous, and simple as a result. If we choose "Yes", then we need to add a way for the user (extra function argument) to specify which columns are in fact bools, and therefore override the reader to parse INT -> BOOL.

2. If we don't decide, and I simply add sqlite to the existing code, and we rely on accepting Bools when supported, it is very easy to end up in the following situation:

Postgres (bool) -> Flux (bool) -> MySQL (int), so far "OK" -> flux (int) oh dear -> Postgres (int) if this is the same DB as originally used, we now have an issue.

So I guess what I'm asking is:
1. Do we support SQL booleans?

2. If Yes, is the following API change acceptable: 
```
sql.from(driverName: , dataSourceName: , query: , boolColumns["bool_as_int_column",...])
```
3. I suggest that boolColumns is optional (most DBs will not be using bools) to not break existing API - but then we need to let the users know clearly, that if their SQL source has a boolean column, and they don't explicitly list it, the query will fail with a type error.

Hopefully this makes sense? I only ask because adding SQLite into the existing code took me less than 4 hours, but I did several days worth of testing, which exposed these issues (I tested Flux against live instances of mysql, influxdb, mysql, and sqlite).  
SQLite (and Postgres) is by nature a very happy little DB that simply stores whatever you want, wherever you want, which forces me to ask you which direction you want to go in.

**Comment by nathanielc** [comment]
@nicgrobler Good question. Flux should not make any type conversions without explicit user input, and the sql code itself should not make those type conversions, rather they should be a separate step in the Flux code. For example ` sql.from() |> ... |> toInt(column:"bool_as_int_column")` etc.

So that means that we should support SQL bools, meaning that the `sql.from` function can produce bool typed columns if the connected database supports it. Meaning the Flux table should use the same types as the SQL table. This mean that we do not need the `boolColumns` parameter to the `sql.from` function, it will be an explicit step after retrieving the data from sql. 

 For `sql.to` it will be an error to write `bool` type data to databases that don't support it. 

In other words the `sql` Flux code should not make any type conversion decisions for the user. 

If a user wants to treat an `int` column for a MySQL table as a boolean then then will need to do that conversion explicitly and make sure to convert back before writing to a MySQL database.

```
sql.from(<mysql database>)
    |> toBool(column:"int_as_bool_column")
    |> ... 
    |> toInt(column: "int_as_bool_column")
    |> sql.to(<mysql database>)
```

Does that answer your question?

**Comment by nicgrobler** [comment]
@nathanielc - perfect. Thanks a lot for taking the time to answer.
I will have the PR to you as soon as I have added some more tests
